Userland/Services/DHCPClient/DHCPv4Client.cpp

```diff
@@ -361,6 +361,11 @@ void DHCPv4Client::dhcp_request(DHCPv4Transaction& transaction, DHCPv4Packet con
     // set packet options
     builder.set_message_type(DHCPMessageType::DHCPRequest);
     builder.add_option(DHCPOption::RequestedIPAddress, sizeof(IPv4Address), &offer.yiaddr());
+
+    auto maybe_dhcp_server_ip = offer.parse_options().get<IPv4Address>(DHCPOption::ServerIdentifier);
+    if (maybe_dhcp_server_ip.has_value())
+        builder.add_option(DHCPOption::ServerIdentifier, sizeof(IPv4Address), &maybe_dhcp_server_ip.value());
```

Comment:
We used to have ServerIdentifier, but ended up removing it for compat with some other dhcp servers, see https://github.com/SerenityOS/serenity/commit/6b360834f2cb679edcb1daa92011ba2238e85242

Reply:
This patch doesn't use DHCP server IP address from the `siaddr` field, but rather the `DHCP Option 54` which is optional and can be null.

In this case, if the DHCP server doesn't give us this option in the DHCPOffer packet then we just don't send any `ServerIdentifier` option back. This should maintain compatibility and allow other servers that require this option to work :)

<img width="838" alt="Screen Shot 2022-04-15 at 2 14 04 AM" src="https://user-images.githubusercontent.com/9163891/163461394-923dbb05-1a45-4afc-aca9-0833a38c586a.png">